crates/holochain_sqlite/src/functions.rs

```diff
@@ -0,0 +1,43 @@
+use kitsune_p2p::dht::hash::RegionHash;
+use num_traits::Zero;
+use rusqlite::{functions::*, *};
+
+pub fn add_custom_functions(conn: &Connection) -> Result<()> {
+    conn.create_aggregate_function(
+        "REDUCE_XOR",
+        -1,
+        FunctionFlags::SQLITE_DETERMINISTIC | FunctionFlags::SQLITE_DIRECTONLY,
+        AggregateXor,
+    )?;
+
+    Ok(())
+}
+
+pub struct AggregateXor;
+
+impl Aggregate<RegionHash, Vec<u8>> for AggregateXor {
+    fn init(&self, _ctx: &mut Context<'_>) -> Result<RegionHash> {
+        Ok(RegionHash::zero())
+    }
+
+    fn step(&self, ctx: &mut Context<'_>, v: &mut RegionHash) -> Result<()> {
+        let blob: Vec<u8> = ctx.get(0)?;
```

Comment:
aww, rust... part of the efficiency of sqlite aggregate function is lost here by requiring the copy of data out of sqlite memory...

but i get it'd be hard to return a `&'? [u8]` here while satisfying rusts safety rules...